pandas/core/frame.py

```diff
@@ -5661,6 +5668,9 @@ def reset_index(
         col_fill : object, default ''
             If the columns have multiple levels, determines how the other
             levels are named. If None then the index name is repeated.
+        allow_duplicates : bool or str, default False
```

Comment:
this is not a good api

Reply:
Above I commented that the previous version

    df.reset_index(allow_duplicates=None)

is not a good API and thought 

   df.reset_index(allow_duplicates="use_flag")

would be better. Do you prefer the first, or another API altogether?

Reply:
Problem solved!
The caller can always access the flag directly if they really care.
`flags.allows_duplicate_labels = False` seems fairly broken anyway due to `inplace `handling (Yes, I will make an Issue).